eclipse-che/che-server
# Support SSH urls

### What does this PR do?
Extend each SCM provider implementation to support SSH urls.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/22354

### How to test this PR?

1. Deploy Che with the PR image: `quay.io/eclipse/che-server:pr-539`
2. [Configure Che to use SSH keys for Git operations](https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#configuring-devworkspaces-to-use-ssh-keys-for-git-operations)
3. Try to start a workspace from an SSH url of a **PUBLIC** repository to avoid configuring OAuth or PAT, see: https://github.com/eclipse-che/che-docs/pull/2639

See: 
* Dashboard shows a notification that the devfile has been fetched:
![screenshot-eclipse-che apps cluster-c8wk5 c8wk5 sandbox2818 opentlc com-2023 08 08-16_06_34](https://github.com/eclipse-che/che-server/assets/7668752/a86379ce-a7e7-47dc-bf66-0363bdf25d49)
* Workspace starts successfully

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
